src/elements/content-sidebar/__tests__/DocGenSidebar.test.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { render, screen, waitFor, fireEvent } from '../../../test-utils/testing-library';
+
+import { DocGenSidebarComponent as DocGenSidebar } from '../DocGenSidebar/DocGenSidebar';
+import mockData from '../__mocks__/DocGenSidebar.mock';
+
+const docGenSidebarProps = {
+    getDocGenTags: jest.fn().mockReturnValue(
+        Promise.resolve({
+            pagination: {},
+            data: mockData,
+        }),
+    ),
+};
+
+const noTagsMock = jest.fn().mockReturnValue(Promise.resolve({ data: [] }));
+const errorTagsMock = jest.fn().mockRejectedValue([]);
+const noDataMock = jest.fn().mockReturnValue(Promise.resolve({}));
+
+const defaultProps = {
+    ...docGenSidebarProps,
+};
+
+describe('elements/content-sidebar/DocGenSidebar', () => {
+    const renderComponent = (props = defaultProps) =>
+        render(<DocGenSidebar logger={{ onReadyMetric: jest.fn() }} {...props} />);
+
+    test('componentDidMount() should call fetch tags', async () => {
+        renderComponent();
+        await waitFor(() => expect(docGenSidebarProps.getDocGenTags).toHaveBeenCalled());
+    });
+
+    test('should render DocGen sidebar component correctly with tags list', async () => {
+        renderComponent();
+        const tagList = await screen.findAllByTestId('bcs-TagsSection');
+        expect(tagList).toHaveLength(2);
+    });
+
+    test('should render empty state when there are no tags', async () => {
+        renderComponent({
+            ...defaultProps,
+            getDocGenTags: noTagsMock,
+        });
+
+        const emptyState = await screen.findByText('This document has no tags');
+        expect(emptyState).toBeInTheDocument();
+    });
+
+    test('should render loading state', async () => {
+        renderComponent({
+            ...defaultProps,
+            getDocGenTags: noTagsMock,
+        });
+
+        const loadingState = await screen.getByRole('status', { name: 'Loading' });
```

Comment:
Is this test reliable? Without a timeout how do we always ensure that the loading indicator will be in the document since the isLoading state gets set to false at the end of the async loadTags function.

Reply:
I'll add a getDocGenTags mock that resolves after a timeout of 1 second

Reply:
addressed